examples/object-final-fields/object_final_fields.bal

```diff
@@ -0,0 +1,97 @@
+import ballerina/io;
+
+type Details record {|
+    int id;
+    string country;
+|};
+
+// A class definition with both `final` and non-`final` fields.
+class Employee {
+    string department;
+    // `details` is a `final` field to which a value cannot be assigned
+    // once the object value is created. However, since the type of `details` is
+    // a mutable type, updates can be attempted on the `details` field.
+    final Details details;
+
+    function init(string department, Details details) {
+        self.department = department;
+        self.details = details;
+    }
+}
+
+// A class definition for a `readonly` object.
+// If the class definition includes `readonly`, all the fields in the
+// object are considered to be `readonly` fields.
+// The effective types of all the fields are the intersection of the
+// specified type and `readonly`.
+readonly class MainController {
+    int id;
+    string[] codes;
+
+    function init(int id, readonly & string[] codes) {
+        self.id = id;
+        // The expected type for `codes` is `readonly & string[]`.
+        self.codes = codes;
+    }
+
+    function getId() returns string {
+        return string `Main: ${self.id}`;
+    }
+}
+
+int currentId = 1000;
+
+public function main() {
+
+    // Define a mutable `Details` value.
+    Details details = {
+        id: 112233,
+        country: "Sri Lanka"
+    };
+
+    // Use `details` as the value for the `final` `details` field in `Employee`.
+    Employee emp = new ("IT", details);
+
+    io:println("ID: ", emp.details.id);
+
+    // Even though the `details` field of `Employee` is `final`,
+    // it can hold a mutable value. Thus, we can attempt updating
+    // the `id` field of the `details` field.
+    Details empDetails = emp.details;
+    empDetails.id = 223344;
```

Comment:
Should be able to do 

```suggestion
    emp.details.id = 223344;
```

But this is not possible atm - https://github.com/ballerina-platform/ballerina-lang/issues/25963.